packages/ts/react-crud/src/autoform-field.tsx

```diff
@@ -1,36 +1,103 @@
+type AutoFormTextFieldProps = SharedFieldProps & TextFieldProps;
+
+function AutoFormTextField({ propertyInfo, form, ...other }: AutoFormTextFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the typing issue
+  return <TextField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormIntegerFieldProps = IntegerFieldProps & SharedFieldProps;
+
+function AutoFormIntegerField({ propertyInfo, form, ...other }: AutoFormIntegerFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <IntegerField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormNumberFieldProps = NumberFieldProps & SharedFieldProps;
+
+function AutoFormDecimalField({ propertyInfo, form, ...other }: AutoFormNumberFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <NumberField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormDateFieldProps = DatePickerProps & SharedFieldProps;
+
+function AutoFormDateField({ propertyInfo, form, ...other }: AutoFormDateFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <DatePicker {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormTimeFieldProps = SharedFieldProps & TimePickerProps;
+
+function AutoFormTimeField({ propertyInfo, form, ...other }: AutoFormTimeFieldProps) {
   const model = getPropertyModel(form, propertyInfo);
-  return <TextField disabled={disabled} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <TimePicker {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
 }
 
-function AutoFormIntegerField({ propertyInfo, form, disabled }: AutoFormFieldProps) {
+type AutoFormDateTimeFieldProps = DateTimePickerProps & SharedFieldProps;
+
+function AutoFormDateTimeField({ propertyInfo, form, ...other }: AutoFormDateTimeFieldProps) {
   const model = getPropertyModel(form, propertyInfo);
-  return <IntegerField disabled={disabled} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <DateTimePicker {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormEnumFieldProps = SelectProps & SharedFieldProps;
+
+function AutoFormEnumField({ propertyInfo, form, ...other }: AutoFormEnumFieldProps) {
+  const model = getPropertyModel(form, propertyInfo) as EnumModel;
+  const options = Object.keys(model[_enum]).map((value) => ({
+    label: value,
```

Comment:
Let's render a more user-friendly label here as mentioned in the [ticket](https://github.com/vaadin/hilla/issues/1528). With the current example model I would expect the following options:
- `Male`
- `Female`
- `Non binary` or `Non Binary`

Reply:
I propose to implement that separately by addressing https://github.com/vaadin/hilla/issues/1617.